code/modules/reagents/reagent_containers/hypospray.dm

```diff
@@ -98,22 +100,16 @@
 	return OXYLOSS//ironic. he could save others from oxyloss, but not himself.
 
 /obj/item/reagent_containers/hypospray/medipen/attack(mob/M, mob/user)
-	if(!reagents.total_volume)
-		to_chat(user, "<span class='warning'>[src] is empty!</span>")
-		return
 	..()
-	if(!iscyborg(user))
-		reagents.maximum_volume = 0 //Makes them useless afterwards
-		reagents.flags = NONE
+	reagents.maximum_volume = 0 //Makes them useless afterwards
+	reagents.flags = NONE
+	update_icon()
+
+/obj/item/reagent_containers/hypospray/medipen/attack_self(mob/user)
+	inject(user, user)
+	reagents.maximum_volume = 0 //Makes them useless afterwards
+	reagents.flags = NONE

```

Comment:
This should be in the inject proc.

Something like:
```
/obj/item/reagent_containers/hypospray/medipen/inject(mob/living/M, mob/user)
	. = ..()
	if(.)
		reagents.maximum_volume = 0 //Makes them useless afterwards
		reagents.flags = NONE
		update_icon()
```
Of course, for this to work, the inject proc would need to return TRUE on sucessful injection which is another thing that should be added.